### PR TITLE
changes to MDE, AAD, and SOAR Essentials solutions

### PR DESCRIPTION
**Pull request by BenjiSec**
Required items, please complete
   
   Change(s):
  Update to
Azure Active Directory solution
- Added Revoke-AADSignInSessions playbooks
- Update to Block-AADUser playbooks

MicrosoftDefenderForEndpoint solution
- added Unisolate-MDEMachine-entity-trigger playbook
- added Isolate-MDEMachine-entity-trigger playbook

SentinelSOARessentials solution
- added workbooks
1. AutomationHealth
2. IncidentOverview
3. IncidentTasksWorkbook
4. SecurityOperationsEfficiency

-added playbooks
1. CreateIncident-MicrosoftForms
2. CreateIncident-SharedMailbox
3. M365D_BEC_Playbook_for_SecOps-Tasks
4. M365D_Phishing_Playbook_for_SecOps-Tasks
5. M365D_Ransomware_Playbook_for_SecOps-Tasks
6. Send-Teams-adaptive-card-on-incident-creation

Deleted playbooks from Playbook folder (moved to Solutions, removing duplicates)
Unisolate-MDEMachine-entity-trigger - added to MDE solution
Isolate-MDEMachine-entity-trigger - added to MDE solution
Revoke-AADSignInSessions - added to AAD solution
Revoke-AADSignInSessions-entityTrigger - added to AAD solution
CreateIncident-MicrosoftForms - added to SOAR Essentials solution
CreateIncident-SharedMailbox - added to SOAR Essentials solution
M365D_BEC_Playbook_for_SecOps-Tasks - added to SOAR Essentials solution

   Reason for Change(s):
   - Merging playbooks with solutions

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @BenjiSec looking into this

**Comment by v-prasadboke** [comment]
Hello @BenjiSec working on this

**Comment by v-prasadboke** [comment]
Hello @rahul0216 and @manishkumar1991 can you please review the playbooks and approve

**Comment by BenjiSec** [comment]
Hi all, when we can expect that this is approved as we are waiting for this to publish blog about new Task support in Microsoft Sentinel?

**Comment by manishkumar1991** [comment]
> Hi all, when we can expect that this is approved as we are waiting for this to publish blog about new Task support in Microsoft Sentinel?

Hello Benjamin ,

We are reviewing your PR .

File: Solutions/Azure Active Directory/Playbooks/Revoke-AADSignInSessions/readme.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+$GraphAppId = "00000003-0000-0000-c000-000000000000"
+$PermissionName = "User.ReadWrite.All" 
+
+$GraphServicePrincipal = Get-AzureADServicePrincipal -Filter "appId eq '$GraphAppId'"
+$AppRole = $GraphServicePrincipal.AppRoles | Where-Object {$_.Value -eq $PermissionName -and $_.AllowedMemberTypes -contains "Application"}
+New-AzureAdServiceAppRoleAssignment -ObjectId $MI.ObjectId -PrincipalId $MI.ObjectId `
+-ResourceId $GraphServicePrincipal.ObjectId -Id $AppRole.Id
+```
+
```
Comment (manishkumar1991) [inline Solutions/Azure Active Directory/Playbooks/Revoke-AADSignInSessions/readme.md:48]: For Playbook to add any comment to incident, it should have assigned "Microsoft Sentinel Responder Role." 

Kindly add the steps for assigning this role in readme.md file

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Isolate-MDEMachine/readme.md [hunk 2]
```diff
@@ -41,4 +47,7 @@ New-AzureAdServiceAppRoleAssignment -ObjectId $MI.ObjectId -PrincipalId $MI.Obje
 ![Incident Trigger](./Isolate-MDEMachine/Isolate-MDEMachine-incident-trigger/images/designerLight.png)
 
 **Alert Trigger**
-![Alert Trigger](./Isolate-MDEMachine/Isolate-MDEMachine-alert-trigger/images/Isolate-MDEMachine_alert.png)>
\ No newline at end of file
+![Alert Trigger](./Isolate-MDEMachine/Isolate-MDEMachine-alert-trigger/images/Isolate-MDEMachine_alert.png)>
+
+**Entity Trigger**
+![Entity Trigger](./Isolate-MDEMachine/Isolate-MDE-Machine-entity-trigger/images/playbookDark.png)>
```
Comment (manishkumar1991) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Isolate-MDEMachine/readme.md:53]: For Playbook to add any comment to incident, it should have assigned "Microsoft Sentinel Responder Role."

Kindly add the steps for assigning this role in readme.md file

File: Solutions/Azure Active Directory/Playbooks/Revoke-AADSignInSessions/readme.md (hunk 1 above)
Comment (BenjiSec) [reply Solutions/Azure Active Directory/Playbooks/Revoke-AADSignInSessions/readme.md:48]: added

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Isolate-MDEMachine/readme.md (hunk 2 above)
Comment (BenjiSec) [reply Solutions/MicrosoftDefenderForEndpoint/Playbooks/Isolate-MDEMachine/readme.md]: added

Comment (BenjiSec) [reply Solutions/MicrosoftDefenderForEndpoint/Playbooks/Isolate-MDEMachine/readme.md]: can we have it approved now as we must publish blog about Tasks - artifacts for it are in SOAR Essential solution.

**Comment by v-prasadboke** [comment]
Hello @manishkumar1991 benjisec has pushed the changes

**Comment by v-prasadboke** [comment]
Hello @manishkumar1991 witing for your feedback

**Comment by BenjiSec** [comment]
Hi @manishkumar1991 - can we please merge this as we are waiting this to update official Microsoft Sentinel docs around Tasks and to publish blog about new possibilities.

**Review by manishkumar1991 — APPROVED** [review]
approved

**Comment by v-prasadboke** [comment]
Thank you @manishkumar1991,
Continuing to review this pr

**Comment by v-prasadboke** [comment]
Hello @BenjiSec please update the metadata of this playbook ' Solutions/Azure Active Directory/Playbooks/Revoke-AADSignInSessions/alert-trigger/azuredeploy.json '. Title & description is missing from the metadata.  
You can refer to this playbook for metadata ' Solutions/Azure Active Directory/Playbooks/Revoke-AADSignInSessions/entity-trigger/azuredeploy.json'. If needed please add other parameters as well with reference to the Playbook metadata.

Title for this playbook ' Solutions/Azure Active Directory/Playbooks/Revoke-AADSignInSessions/incident-trigger/azuredeploy.json ' is " Revoke-AADSignInSessions " If its ok can we change it to " Revoke AAD Sign-in session using incident trigger "

**Comment by v-prasadboke** [comment]
And parameter of this two playbooks metadata is set to null " Solutions/MicrosoftDefenderForEndpoint/Playbooks/Isolate-MDEMachine/Isolate-MDE-Machine-entity-trigger/azuredeploy.json " & " Solutions/MicrosoftDefenderForEndpoint/Playbooks/Unisolate-MDEMachine/Unisolate-MDE-Machine-entity-trigger/azuredeploy.json " 

Is this property null or can be filled with some instructions / Data 

image

**Comment by v-prasadboke** [comment]
Hello @BenjiSec please add Workbook metadata of the Workbooks of ' SentinelSOARessentials ' in the given file path 
" Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json "

**Comment by malowe101** [comment]
@v-prasadboke I deployed the workbook and confirmed that it works fine. Please make sure that there is an incident number entered within the workbook to load the widgets and content.

image

Regarding the addition of the workbooks to the workbooksmetadata file, the 4 workbooks in this solution are all existing workbooks in the gallery. Is it still required that they are added if they are in a solution?

**Comment by v-prasadboke** [comment]
> Regarding the addition of the workbooks to the workbooksmetadata file, the 4 workbooks in this solution are all existing workbooks in the gallery. Is it still required that they are added if they are in a solution?

Hello @BenjiSec after repackaging the solution and doing arm-ttk on the maintemplate I came to know the the workbook metadata is not present in the workbookmetadata file. Which is causing an error. 
image

**Comment by v-prasadboke** [comment]
Hello @BenjiSec any updates

**Comment by v-prasadboke** [comment]
Hello @malowe101 There are changes requested above, Thank you.

**Comment by v-prasadboke** [comment]
> Regarding the addition of the workbooks to the workbooksmetadata file, the 4 workbooks in this solution are all existing workbooks in the gallery. Is it still required that they are added if they are in a solution?

Yes @malowe101, Even if the workbooks of ' Sentinel Soar Essentials ' already existed in the gallery but for AutomationHealth 
& IncidentTasksWorkbook workbook metadata is missing in the   " Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json " file.

**Comment by BenjiSec** [comment]
> Azure.Azure-Sentinel (JsonFileValidation)

Hi @v-prasadboke - I tried adding these 2 workbooks to Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json " file, but I'm getting errors. Can you please help with this?

File: Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json [hunk 3]
```diff
@@ -1,5069 +1,5230 @@
...
+            "ThreatIntelligenceIndicator"
+        ],
+        "dataConnectorsDependencies": [
+            "AzureActivity",
+            "SecurityEvents",
+            "Office365",
+            "AzureActiveDirectory",
+            "ThreatIntelligence",
+            "ThreatIntelligenceTaxii",
+
```
Comment (v-prasadboke) [inline Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json:5195]: please remove ' , ' after the curly brace

Comment (BenjiSec) [reply Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json]: Hi, all done now.

Comment (BenjiSec) [reply Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json]: for playbbooks
AAD Revoke SignIn session - metadata is added
For Isolate and Unisolate MDE Machine - entity trigger - metadata is filled properly. I normally don't delete empty values but leave them if something needs to be added later.

**Review by v-prasadboke — APPROVED** [review]

**Comment by v-prasadboke** [comment]
All looks good

**Pull request merged**